pkg/reconciler/ducktype/controller.go

```diff
@@ -35,10 +36,13 @@ func NewController(
 	logger := logging.FromContext(ctx)
 
 	ducktypeInformer := ducktypeinformer.Get(ctx)
+	crdInformer := crdinformer.Get(ctx)
 
 	// TODO: watch CRDs, etc.
 
-	r := &Reconciler{}
+	r := &Reconciler{
+		crdLister: crdInformer.Lister(),
+	}
 	impl := ducktypereconciler.NewImpl(ctx, r)
 
 	logger.Info("Setting up event handlers.")
```

Comment:
We will also need to resync the entire keyspace when a CRD changes. 

So unlike https://github.com/knative/eventing/blob/5863d54a532bd1977b9ff6b506bbd4a72f723c39/pkg/reconciler/source/crd/controller.go#L62 we can not do this directly, we do not have a special label nor own the CRD from the DuckType.

Look around at the global resync stuff we are doing with Brokers and Triggers to get an idea of what I am talking about:

https://github.com/knative/eventing/blob/5863d54a532bd1977b9ff6b506bbd4a72f723c39/pkg/reconciler/mtbroker/controller.go#L139-L151

We can do this as a followup.